src/main/java/stsjorbsmod/patches/RenderBurningBlockPatch.java

```diff
@@ -27,8 +36,47 @@ public static void patch(AbstractCreature __this, SpriteBatch sb, float x, float
                         64.0F,
                         Settings.scale,
                         Settings.scale,
-                        0.0F);
+                        burningRotation);
+                burningRotation += 0.1;
```

Comment:
I wouldn't recommend incrementing by a constant value every re-render like this; there's a lot of variation in how long it takes between render calls, both between different users/machines and different frames on the same machine. I also wouldn't recommend tying it to a single static variable like this, since that will make the rotation speed dependent on the number of burning creatures this is rendering for.

I would recommend the pattern that `AbstractMemory` uses for rotating the active memory effect. The pattern looks like:
* store a rotation timer as a field on the entity in question (here, as a `SpireField` on `AbstractCreature`
* on each *update* tick (not each render), do something like this:
  ```java
    private void updateYourRotationTimer(AbstractCreature c) {
        yourRotationField.set(c, yourRotationField.get(c) - Gdx.graphics.getDeltaTime());
        if (yourRotationField.get(c) < 0.0F) {
            yourRotationField.set(c, YOUR_ROTATION_DURATION);
        }
    }
  ```
* In the render path, use a value of `(yourRotationField.get(c) / YOUR_ROTATION_DURATION) * 360F` for the actual rotation amount to pass to the drawing routines

Reply:
Admittedly, the rotation thing was wholly for my amusement. I'm not totally convinced animating the power icon makes much sense. I'll see what I can do though.